src/test/java/org/arquillian/tests/pom/pageObjects/BlogPage.java

```diff
@@ -22,7 +22,11 @@
     @FindBy(css = "[class = 'tag-cloud']")
     private CloudTagFragment cloudTag;
 
-    public BlogFragment blogContent() {
+    public BlogFragment releaseBlog() {
+        return blogContent;
+    }
+
+    public BlogFragment nonReleaseBlog() {
         return blogContent;
```

Comment:
I don't understand why we have two methods returning the same object. I think the original problem I was suggesting to fix was that on the page were we have multiple blogs calling `blogContent` method will led to the first blog, but that was not obvious from the test itself.

Reply:
I am confused. We check for all the blogs on the blog page and not just the first blog.

Reply:
If that is the case, then we have to rework the structure of page objects we have in place. Not sure if `BlogFragment` should represent more than one blog entry.

Reply:
That makes sense but then don't we check for all the release blogs for release notes or just the first one ?

Reply:
We can always check for all, just "reverse" the logic for it somehow. We anyway never check all, but first `n` displayed, right?

Reply:
As this page is changing over the time I am not sure the test as we have now is really what we want. I would rather think of having two separated test cases.

1. Go to "release" posts and check for release notes in all displayed on the first page, because in there we know that we should have release notes. (plus title and author)

2. Go to "non-release" blog posts and check if title and author are there?

Reply:
> I would rather think of having two separated test cases.

> Go to "release" posts and check for release notes in all displayed on the first page, because in there we know that we should have release notes. (plus title and author)

> Go to "non-release" blog posts and check if title and author are there?

That's what I tried doing here, may be some more refactoring should help.